src_c/transform.c

```diff
@@ -593,6 +592,53 @@ surf_scale(PyObject *self, PyObject *args, PyObject *kwargs)
         return (PyObject *)pgSurface_New(newsurf);
 }
 
+static PyObject*
+surf_scale_by(PyObject *self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *surfobj;
+    PyObject *surfobj2 = NULL;
+    PyObject *new_args = NULL;
+    PyObject *scale_obj = NULL;
+    PyObject *scaley_obj = Py_None;
+    float scale, scaley;
+    SDL_Surface *surf;
+    int width, height;
+    static char *keywords[] = {"surface", "scale", "scale_y", "dest_surface", NULL};
```

Comment:
Ah, I forgot to respond 😅

Thank you for the feedback.

If there was `scale_x` and `scale_y`, users might interpret them as defaulting to 1. So `scale_y=5` should stretch vertically 5x and do nothing to the x.

I would use them positional-only, but I was trying to fit with a big push to get pygame to accept kwargs in #808.

I guess anything could be potentially confusing, it would be up to the docs to set things straight.

Reply:
> If there was scale_x and scale_y, users might interpret them as defaulting to 1.

Ok, I agree.

> I was trying to fit with a big push to get pygame to accept kwargs in #808.

Sure, but this seems like a good place to deviate and go positional-only.

Firstly because `scale` is overloaded as discussed. Secondly because the function name is `scale_by(...)`, the question is "scale by WHAT". `scale_by(factor)` reads as English, so does `scale_by(xscale, yscale)`. `scale_by(scale=scale)` doesn't read as well. `scale_by(dest_surface=screen, scale=5)` is right out.